nevergrad/functions/stsp/core.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Based on a discussion at Dagstuhl's seminar on Computational Intelligence in Games with:
+# - Dan Ashlock
+# - Chiara Sironi
+# - Guenter Rudolph
+# - Jialin Liu
+
+from math import pi, cos, sin
+from typing import Any
+from typing import List
+import matplotlib.pyplot as plt
+import numpy as np
+from ... import instrumentation as inst
+from ...instrumentation.multivariables import Instrumentation
+
+
+# pylint: disable=too-many-instance-attributes,too-many-arguments,too-many-statements,too-many-locals
+class STSP(inst.InstrumentedFunction):
+
+    def __init__(self, seed: int = 0, the_dimension: int = 500) -> None:
+        state = np.random.get_state()
+        np.random.seed(seed)
```

Comment:
You should first create the instrumentation, then use the instrumentaiton random state:
instrumentation.random_state.normal etc...

Reply:
done!